.gitlab-ci.yml

```
@@ -233,21 +238,26 @@ publish:s3:
     - export AWS_ACCESS_KEY_ID=$PUBLISH_AWS_ACCESS_KEY_ID
     - export AWS_SECRET_ACCESS_KEY=$PUBLISH_AWS_SECRET_ACCESS_KEY
 
+    - IMAGE_VERSION=${MENDER_CONVERT_PUBLISH_VERSION:-${CI_COMMIT_REF_NAME}}
+
     - for RASPBERRYPI_PLATFORM in raspberrypi3 raspberrypi4; do
-    -   PUBLISH_NAME=${RASPBIAN_NAME}-${RASPBERRYPI_PLATFORM}-mender-${MENDER_CLIENT_VERSION}.img.xz
+    -   PUBLISH_NAME=${RASPBIAN_NAME}-${RASPBERRYPI_PLATFORM}-mender-convert-${IMAGE_VERSION}.img.xz
     -   echo "Publishing ${PUBLISH_NAME} version to S3"
     -   aws s3 cp ${RASPBERRYPI_PLATFORM}/${RASPBIAN_NAME}-${RASPBERRYPI_PLATFORM}-mender.img.xz
           s3://$S3_BUCKET_NAME/${RASPBIAN_NAME}/arm/${PUBLISH_NAME}
     -   aws s3api put-object-acl --acl public-read --bucket $S3_BUCKET_NAME
           --key ${RASPBIAN_NAME}/arm/${PUBLISH_NAME}
 
-    -   PUBLISH_NAME=${RASPBIAN_NAME}-${RASPBERRYPI_PLATFORM}-mender-${MENDER_CLIENT_VERSION}.mender
+    -   PUBLISH_NAME=${RASPBIAN_NAME}-${RASPBERRYPI_PLATFORM}-mender-convert-${IMAGE_VERSION}.mender
     -   echo "Publishing ${PUBLISH_NAME} version to S3"
     -   aws s3 cp ${RASPBERRYPI_PLATFORM}/${RASPBIAN_NAME}-${RASPBERRYPI_PLATFORM}-mender.mender
           s3://$S3_BUCKET_NAME/${RASPBIAN_NAME}/arm/${PUBLISH_NAME}
     -   aws s3api put-object-acl --acl public-read --bucket $S3_BUCKET_NAME
           --key ${RASPBIAN_NAME}/arm/${PUBLISH_NAME}
     - done
 
-  only:
-    - /^(master|[0-9]+\.[0-9]+\.x)$/
+  rules:
+    - if: '$CI_COMMIT_REF_NAME =~ /^(master|[0-9]+\.[0-9]+\.(x|[0-9]+))$/ && $MENDER_CONVERT_PUBLISH_VERSION == ""'
+      when: on_success
+    - if: '$CI_COMMIT_REF_NAME =~ /^(master|[0-9]+\.[0-9]+\.(x|[0-9]+))$/ && $MENDER_CONVERT_PUBLISH_VERSION != ""'
+      when: manual
```

The rules look okey. An alternative (simplified?) way could be:
```suggestion
  rules:
    - if: '$MENDER_CONVERT_PUBLISH_VERSION != ""'
      when: manual
    - if: '$CI_COMMIT_REF_NAME =~ /^(master|[0-9]+\.[0-9]+\.(x|[0-9]+))$/'
      when: on_success
```

It allows to manually publish from any branch. But maybe could come in handy to publish some candidate to even ask a user to try it out or something. The point is that once we are overriding the version string to publish, I don't see why we need to restrict it to master and release branches.

---

True, this is an unnecessary restriction. I'll remove it.

---

I removed it, but replaced it with an expression forbidding `pr_*` references. This has no practical effect, since you can't publish from those anyway (missing keys), but is for convenience: If you forget about it, you may spend two hours waiting for tests to pass, only to find out you can't publish afterwards. With the that expression it will be immediately obvious from the pipeline graph.